src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -479,5 +488,12 @@ internal override int LastIndexOf(T[] array, T value, int startIndex, int count)
             }
             return -1;
         }
+
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            // The LongEnumEqualityComparer does not exist on 4.0 so we need to serialize this comparer as ObjectEqualityComparer
```

Comment:
It doesn't matter. Because this code exists in 4.7, the type in 4.7  doesn't have a deserialization ctor, which means core has to do the same dance. The comment could be tweaked if desired, though.